packages/teams-js/src/public/geoLocation.ts

```diff
@@ -0,0 +1,120 @@
+import { sendAndHandleSdkError as sendAndHandleError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { DevicePermission, ErrorCode } from './interfaces';
+import { runtime } from './runtime';
+
+/**
+ * Namespace to interact with the geoLocation module-specific part of the SDK. This is the newwer version of location module.
+ */
+export namespace geoLocation {
+  /**
+   * Data struture to represent the location information
+   */
+  export interface Location {
+    /**
+      Latitude of the location
+      */
+    latitude: number;
+    /**
+      Longitude of the location
+      */
+    longitude: number;
+    /**
+      Accuracy of the coordinates captured
+      */
+    accuracy?: number;
+    /**
+      Time stamp when the location was captured
+      */
+    timestamp?: number;
+  }
+  /**
+   * Fetches current user coordinates
+   * @returns Promise of user's current {@linkcode Location}
+   */
+  export function getCurrentLocation(): Promise<Location> {
+    ensureInitialized(FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    return sendAndHandleError('location.getLocation', { allowChooseLocation: false, showMap: false });
+  }
+
+  /**
+   * Checks whether or not location has user permission
+   *
+   * @returns if the location has user permission
+   */
+  export function hasPermission(): Promise<boolean> {
+    ensureInitialized(FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.GeoLocation;
+
+    return new Promise<boolean>(resolve => {
+      resolve(sendAndHandleError('permissions.has', permissions));
+    });
+  }
+
+  /**
+   * Request user permission for location
+   *
+   * @returns if the user conseted permission for location
+   */
+  export function requestPermission(): Promise<boolean> {
+    ensureInitialized(FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.GeoLocation;
+
+    return new Promise<boolean>(resolve => {
+      resolve(sendAndHandleError('permissions.request', permissions));
+    });
+  }
+
+  export function isSupported(): boolean {
+    return runtime.supports.geoLocation ? true : false;
+  }
+
+  /**
+   * Namespace to interact with the location on map module-specific part of the SDK.
+   */
+  export namespace map {
+    /**
+     * Allows user to choose location on map
+     *
+     * @returns The location chosen by the user after closing the map
```

Comment:
How about: `@returns Promise that will resolve with the location chosen by the user` (is the *after closing the map* necessary to the function of this API, or is it a UX detail that could change in the future?)